SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -7363,80 +7363,92 @@ void CEulerSolver::BC_Supersonic_Inlet(CGeometry *geometry, CSolver **solver_con
    so all flow variables can be imposed at the inlet.
    First, retrieve the specified values for the primitive variables. ---*/
 
-  const su2double Temperature = config->GetInlet_Temperature(Marker_Tag) / config->GetTemperature_Ref();
-  const su2double Pressure = config->GetInlet_Pressure(Marker_Tag) / config->GetPressure_Ref();
-  const auto* Vel = config->GetInlet_Velocity(Marker_Tag);
+  /*--- Loop over all the vertices on this boundary marker ---*/
 
-  su2double Velocity[MAXNDIM] = {0.0};
-  for (unsigned short iDim = 0; iDim < nDim; iDim++)
-    Velocity[iDim] = Vel[iDim] / config->GetVelocity_Ref();
+  SU2_OMP_FOR_DYN(OMP_MIN_SIZE)
+  for (auto iVertex = 0ul; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+    const auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
 
-  /*--- Density at the inlet from the gas law ---*/
+    if (!geometry->nodes->GetDomain(iPoint)) continue;
 
-  const su2double Density = Pressure / (Gas_Constant * Temperature);
+    /*--- Retrieve the specified conditions for this inlet. ---*/
 
-  /*--- Compute the energy from the specified state ---*/
+    const su2double Temperature = config->GetInlet_Temperature(Marker_Tag) / config->GetTemperature_Ref();
+    const su2double Pressure = config->GetInlet_Pressure(Marker_Tag) / config->GetPressure_Ref();
+    const auto* Vel = config->GetInlet_Velocity(Marker_Tag);
 
-  const su2double Velocity2 = GeometryToolbox::SquaredNorm(int(MAXNDIM), Velocity);
-  su2double Energy = Pressure / (Density * Gamma_Minus_One) + 0.5 * Velocity2;
-  if (tkeNeeded) Energy += GetTke_Inf();
+    su2double Velocity[MAXNDIM] = {0.0};
```

Comment:
Should these variables be inside the for loop? They are config values and don't vary between verticies.

Reply:
I am not that sure about here, since they contain velocity, and velocity will change between vertices. Actually I didn't change the structure of the loop here if you take a look at the original code. But we had better to ask the developers to take a look. @pcarruscag @bigfooted